docs/api.rst

```diff
@@ -144,6 +145,105 @@ Variable recombination rates
 .. autoclass:: msprime.RecombinationMap
     :members:
 
+.. _sec_api_simulation_models:
+
+*****************
+Simulation models
+*****************
+
+The default simulation model in ``msprime`` is the standard coalescent with recombination
+model. We also support a number of different models, which are documented in this section.
+
+Simulations models are specified using the ``model`` parameter to
+:func:`.simulate`. This parameter can either take the form of a
+string describing the model (e.g. ``model="smc"``) or an instance of a
+model definition class (e.g ``model=msprime.SmcApproxCoalescent(1)``).
+The available models are documented in the following subsections.
+
+A key element of simulation models in ``msprime`` is the concept
+of a "reference population size". Simulation models such as
+the coalescent are defined
+in terms of "scaled time". Time in the standard diploid coalescent
+is measured in units of :math:`4 N_e` generations, and
+one of the ways in which msprime tries to make life easier for
+users is to automatically convert times and rates to and
+from units of generations. Thus, a key responsibility for a simulation model is to
+convert times specified by users in generations into "model time"
+(in which the simulation is performed) and to tranlate the
+simulated model times back into generations. This is what the
+reference population size associated with a model is used for
+and fundamentally what it means. The ``Ne`` argument to
+:func:`simulate` can also be used to define this parameter
+when combined with a string shorthand for a model.
+Thus
+
+.. code-block:: python
+
+    msprime.simulate(10, Ne=1000, model="smc")
+
+and
+
+.. code-block:: python
+
+    msprime.simulate(10, model=msprime.SmcApproxCoalescent(1000))
+
+define the same simulation.
+
+The situation is somewhat confused by the sizes associated
+with specific populations using, e.g., the
+:class:`.PopulationConfiguration` class. In coalescent models, these
+sizes are used to compute rates of coalescence, and have no direct
+relationship to the model's reference population size. Thus, we may
+have several populations, all of which have different sizes and none
+equal to the model's reference population size.
+The model's reference population size is used to scale all times
+and rates, irrespective of the sizes of the various individual
+populations.
+
+The situation for the :class:`.DiscreteTimeWrightFisher` model
+is different. In this case, there is no concept of rescaling time
+as time is always measured in generations. Therefore, in this case the
+reference population size has no function and is just used as a
```

Comment:
This isn't quite true as it's used in the population growth calculations, I need to fix this

Reply:
also the discussion of the model reference size being divorced from the actual population sizes might throw users. perhaps better to suggest that the user choose a "sensible reference", like the population size of pop0 at time zero?

Reply:
Isn't it true for the coalescent also that the reference size is a shortcut for specifying individual population sizes? Like, I think that `Ne` is *not* an overall time scaling, because setting population sizes in population configurations overrides it?

Reply:
Ne is *definitely* used for scaling and unscaling times @petrelharp -- see [here](https://github.com/tskit-dev/msprime/blob/master/lib/msprime.c#L4219)